### PR TITLE
fix malformed packet when depacketizing a tecmp message

### PR DESCRIPTION
**Pull request by rawiamoalla**

### CODE REVIEW

File: apps/app.cpp [hunk 1]
```diff
@@ -16,40 +16,40 @@ using namespace std;
...
-		}
+	case TECMP_TYPE_CM_STATUS:
+	{
+		tecmp_cm_status cm_status;
+		tecmp_get_cm_status(header, data, &cm_status);
+		auto v = cm_status.vendor;
+		printf("CM %u:\n", cm_status.cm_id);
+		printf("\t Version: v%u.%u.%u\n", v.sw_version_major, v.sw_version_minor, v.sw_version_patch);
+		break;
+	}
```
Comment (kayoub5) [inline apps/app.cpp:27]: @rawiamoalla please revert indent changes

File: apps/app.cpp [hunk 2]
```diff
@@ -106,32 +106,20 @@ int main(int argc, char* argv[]) {
...
 
-				if (header.data_type == TECMP_DATA_ETHERNET)
-				{
-					// Ethernet Frame, write it back to the PCAP file without TECMP header
-					newP = RawPacket(data, header.length, time, false);
-				}
-				else
-				{
-					// Print Status to stdout
-					print_status(header, data);
```
Comment (kayoub5) [inline apps/app.cpp:118]: print_status not used anymore

File: apps/app.cpp [hunk 3]
```diff
@@ -140,11 +87,10 @@ int main(int argc, char* argv[]) {
 				res = tecmp_next(p.getRawData(), p.getRawDataLen(), &iterator, &header, &data);
 			}
 		}
-	}
 
-	reader->close();
-	writer.close();
+		reader->close();
+		writer.close();
```
Comment (kayoub5) [inline apps/app.cpp:92]: Bad indent

File: apps/app.cpp [hunk 4]
```diff
@@ -140,11 +87,10 @@ int main(int argc, char* argv[]) {
...
 
-	// free reader memory because it was created by IFileReaderDevice::getReader()
-	delete reader;
-}
+		// free reader memory because it was created by IFileReaderDevice::getReader()
+		delete reader;
+	}
```
Comment (kayoub5) [inline apps/app.cpp:96]: bad indent

**Comment by rawiamoalla** [comment]
@kayoub5 LGTY ?

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**